From 2f62ec53fe143baf3b3d359387a2a1c2fe28d6de Mon Sep 17 00:00:00 2001
From: Dongwon Kim <dongwon.kim@intel.com>
Date: Mon, 28 Jul 2025 15:41:46 -0700
Subject: [PATCH 1/3] hw/display/virtio-gpu: Initialize dmabuf_fd for the blob
 resource with -1

0 of dmabuf_fd is supposed to be a valid file desriptor value so it can't
be its initial value before file is open.

Signed-off-by: Dongwon Kim <dongwon.kim@intel.com>
---
 hw/display/virtio-gpu.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hw/display/virtio-gpu.c b/hw/display/virtio-gpu.c
index cf84c54ac3..76a594973c 100644
--- a/hw/display/virtio-gpu.c
+++ b/hw/display/virtio-gpu.c
@@ -368,6 +368,7 @@ static void virtio_gpu_resource_create_blob(VirtIOGPU *g,
     res = g_new0(struct virtio_gpu_simple_resource, 1);
     res->resource_id = cblob.resource_id;
     res->blob_size = cblob.size;
+    res->dmabuf_fd = -1;
 
     ret = virtio_gpu_create_mapping_iov(g, cblob.nr_entries, sizeof(cblob),
                                         cmd, &res->addrs, &res->iov,
-- 
2.43.0

